docs/handbook/04_streams.md

```diff
@@ -127,13 +127,31 @@ Content-Type: text/vnd.turbo-stream.html; charset=utf-8
 
 This `messages/message` template partial can then also be used to re-render the message following an edit/update operation. Or to supply new messages created by other users over a WebSocket or a SSE connection. Being able to reuse the same templates across the whole spectrum of use is incredibly powerful, and key to reducing the amount of work it takes to create these modern, fast applications.
 
-
 ## Progressively Enhance When Necessary
 
 It's good practice to start your interaction design without Turbo Streams. Make the entire application work as it would if Turbo Streams were not available, then layer them on as a level-up. This means you won't come to rely on the updates for flows that need to work in native applications or elsewhere without them.
 
 The same is especially true for WebSocket updates. On poor connections, or if there are server issues, your WebSocket may well get disconnected. If the application is designed to work without it, it'll be more resilient.
 
+If a Turbo Drive visit needs to integrate with Turbo Streams, you can overwrite its [Accept][] header within a `turbo:before-fetch-request` event listener:
+
+```js
+let specialCaseRequest = false
```

Comment:
as a Stimulus controller:

```js
import { Controller } from "stimulus"

export default class extends Controller {
  initialize() {
    this.handleFetch = this.handleFetch.bind(this)
  }

  connect() {
    addEventListener("turbo:before-fetch-request", this.handleFetch, true)
  }

  disconnect() {
    removeEventListener("turbo:before-fetch-request", this.handleFetch, true)
  }

  registerClick() {
    this.nextRequestIsStream = true
  }

  handleFetch({ detail }) {
    if (this.nextRequestIsStream) {
      const fetchOptions = detail.fetchOptions
      if (fetchOptions) {
        const { headers } = fetchOptions
        headers.Accept = [ "text/vnd.turbo-stream.html", headers.Accept ].join(", ")
      }
    }

    this.nextRequestIsStream = false
  }
}
```

Reply:
@ghiculescu thank you for drafting that up. I was hoping that a set of event listeners would boil down to the core concepts at play without the Stimulus structural debris.

If you don't mind, here's an even more slim version of your example that listens at the `<body>` level and opts into submissions via a `data-*-target` attribute:

```html
<body data-controller="turbo-streams" data-action="turbo:click->turbo-streams#capture turbo:before-fetch-request@document->turbo-streams#injectHeaders">
  <!-- ... -->
  <a href="/foo" data-turbo-streams-target="link">Load content via Turbo Stream</a>
</body>
```

the controller:

```js
// app/javascript/controllers/turbo_streams_controller.js
import { Controller } from "stimulus"

export default class extends Controller {
  static targets = [ "link" ]

  capture({ target }) {
    this.clickedLink = this.linkTargets.find(link => link === target)
  }

  injectHeaders({ detail }) {
    if (this.clickedLink) {
      const { headers } = detail.fetchOptions || {}
      if (headers) {
        headers.Accept = [ "text/vnd.turbo-stream.html", headers.Accept ].join(", ")
      }
    }

    this.clickedLink = null
  }
}
```

Reply:
Nice. That controller is even better!

Reply:
@ghiculescu @seanpdoyle A slight variation of this controller without `targets`. Check this out:

```javascript
import { Controller } from "stimulus"

export default class extends Controller {
  // Actions

  inject() {
    addEventListener("turbo:before-fetch-request", this.injectHeaders, { once: true })
  }

  // Private

  injectHeaders({ detail }) {
    const { headers } = detail.fetchOptions || {}
    if (headers) {
      headers.Accept = ["text/vnd.turbo-stream.html", headers.Accept].join(", ")
    }
  }
}
```

Now you can just use this:

```html
<body data-controller="turbo-streams">
  <a href="/foo" data-action="turbo:click->turbo-streams#inject">Load content via Turbo Stream</a>
</body>
```

Reply:
This is really slick! Thanks for sharing the `addEventListener(..., { once: true })` pattern @rainerborene, it's much more elegant than reference tracking.